### PR TITLE
Implements random npc movement via a loopable queueable task

### PR DESCRIPTION
**Pull request by charles-m-knox**
In order to implement random NPC movements, I created a new task called `QueueableTask`.  It is a general-purpose task that executes a callback on each tick, and also optionally allows a base task to be triggered on the next tick.

I used this to implement `actor.moveSomewhere()` on each tick.

Please provide any thoughts in the implementation. I have only been introduced to this code base as of yesterday, so there are likely some subtleties and nuances that I have glossed over in my implementation. There might even already be an existing task/pipe/etc that implements what I implemented here, and if so, let me know.

### CODE REVIEW

File: src/engine/world/actor/actor.ts [hunk 1]
```diff
@@ -333,13 +335,19 @@ export abstract class Actor {
...
-        // this used to use `setInterval` but will need rewriting to be synced with ticks
-        // see https://github.com/runejs/server/issues/417
-        // this.randomMovementInterval = setInterval(() => this.moveSomewhere(), 1000);
+        this.enqueueBaseTask(new QueueableTask([], this, () => {
+            this.moveSomewhere();
+            return {
+                callbackResult: true,
+                shouldContinueLooping: true,
+            };
+        }, null, { cacheOriginal: false, objectConfig: this as unknown as ObjectConfig })) // TODO: this needs to be better
```
Comment (charles-m-knox) [inline src/engine/world/actor/actor.ts]: Any suggestions on how I might pass in the actor (or just simply nothing) as an ObjectConfig here? I used an escape hatch in TS to get around the type checker, and I would prefer a better solution.

File: src/engine/world/actor/npc.ts [hunk 2]
```diff
@@ -212,7 +212,7 @@ export class Npc extends Actor {
         if(this.metadata.following) {
             return false;
         }
-        return this.updateFlags.faceActor === undefined && this.updateFlags.animation === undefined;
+        return this.updateFlags.faceActor === null && this.updateFlags.animation === null;
```
Comment (charles-m-knox) [inline src/engine/world/actor/npc.ts:215]: This needed to be changed in order to allow NPC's to move. I evaluated their state during regular idle gameplay and these values were `null`. I am not familiar with the purpose of these so I may be breaking something here - please provide thoughts.

**Review by charles-m-knox — COMMENTED** [review]
Left a few notes on areas I'd like to have reviewed.

File: src/engine/world/actor/actor.ts (hunk 1 above)
Comment (Promises) [reply src/engine/world/actor/actor.ts]: made it possible to null this field

File: src/engine/world/actor/npc.ts (hunk 2 above)
Comment (Promises) [reply src/engine/world/actor/npc.ts:215]: nulling instead of setting undefined here seems better, since initial is null

**Review by Promises — APPROVED** [review]

**Pull request merged**